controllers/argocd/repo_server.go

```diff
@@ -586,8 +587,19 @@ func getRepoServerContainerImage(cr *argocdoperatorv1beta1.ArgoCD) string {
 			defaultTag = true
 		}
 	}
-	if e := os.Getenv(common.ArgoCDImageEnvName); e != "" && (defaultTag && defaultImg) {
-		return e
+	if e := os.Getenv(common.ArgoCDImageEnvName); e != "" {
+		if defaultImg && !defaultTag {
+			image, _, found := strings.Cut(e, "@")
+			if found {
+				return argoutil.CombineImageTag(image, tag)
+			}
+			image, _, found = strings.Cut(e, ":")
+			if found {
+				return argoutil.CombineImageTag(image, tag)
+			}
+		} else if defaultImg && defaultTag {
+			return e
+		}
```

Comment:
Is it possible to move this to a common generic function?

I was thinking we could simplify the functions like `getRepoServerContainerImage`, `getApplicationSetContainerImage` and others. Right now they use booleans to decide the `image` and `tag`. We could simplify the logic by setting the image and tag to default values first and then override them if environment variables or values in the CR are set. Basically following:
```
image, tag = default < env < CR
```